core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts

```diff
@@ -1,26 +1,172 @@
 import { MonacoEditorModule } from '@materia-ui/ngx-monaco-editor';
 
-import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
+import { AsyncPipe, NgClass, NgIf, NgSwitch, NgSwitchCase } from '@angular/common';
+import {
+    ChangeDetectionStrategy,
+    Component,
+    ElementRef,
+    EventEmitter,
+    Input,
+    OnInit,
+    Output,
+    ViewChild
+} from '@angular/core';
 
 import { ButtonModule } from 'primeng/button';
 import { DialogModule } from 'primeng/dialog';
 
-import { DotMessageService } from '@dotcms/data-access';
+import { skip } from 'rxjs/operators';
+
+import { DotCMSTempFile } from '@dotcms/dotcms-models';
+import {
+    DotDropZoneComponent,
+    DotMessagePipe,
+    DotSpinnerModule,
+    DropZoneFileEvent
+} from '@dotcms/ui';
+
+import { DotUiMessageComponent } from './components/dot-ui-message/dot-ui-message.component';
+import {
+    BINARY_FIELD_MODE,
+    BINARY_FIELD_STATUS,
+    DotBinaryFieldStore
+} from './store/binary-field.store';
 
 @Component({
     selector: 'dotcms-binary-field',
     standalone: true,
-    imports: [ButtonModule, DialogModule, MonacoEditorModule],
+    imports: [
+        NgIf,
+        NgClass,
+        NgSwitch,
+        NgSwitchCase,
+        AsyncPipe,
+        ButtonModule,
+        DialogModule,
+        DotDropZoneComponent,
+        MonacoEditorModule,
+        DotMessagePipe,
+        DotUiMessageComponent,
+        DotSpinnerModule
+    ],
+    providers: [DotBinaryFieldStore],
     templateUrl: './binary-field.component.html',
     styleUrls: ['./binary-field.component.scss'],
     changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class BinaryFieldComponent {
-    private readonly dotMessageService = inject(DotMessageService);
-    labelCode = this.dotMessageService.get('contenttypes.content.edit.write.code');
-    visible = false;
+export class DotBinaryFieldComponent implements OnInit {
+    //Inputs
+    private _accept: string[] = [];
+    @Input() set accept(accept: string) {
+        this._accept = accept.split(',').map((type) => type.trim());
+    }
+
+    @Input() maxFileSize: number;
+    @Input() helperText: string;
+
+    @ViewChild('inputFile') inputFile: ElementRef;
+
+    @Output() tempFile = new EventEmitter<DotCMSTempFile>();
+
+    readonly dialogHeaderMap = {
+        [BINARY_FIELD_MODE.URL]: 'dot.binary.field.dialog.import.from.url.header',
+        [BINARY_FIELD_MODE.EDITOR]: 'dot.binary.field.dialog.create.new.file.header'
+    };
+    readonly BINARY_FIEL_STATUS = BINARY_FIELD_STATUS;
+    readonly BINARY_FIELD_MODE = BINARY_FIELD_MODE;
+    readonly mode$ = this.dotBinaryFieldStore.mode$;
+    readonly vm$ = this.dotBinaryFieldStore.vm$;
+
+    dotTempFile: DotCMSTempFile;
+    dialogOpen = false;
+    dropZoneActive = false;
```

Comment:
I see we have an store. Can we manage this from the store?

Reply:
@fmontes what do you think?

Reply:
Better move to the store. Use a a view model to use from there.

Reply:
Yes

Reply:
Done!